src/main/java/uk/gov/hmcts/darts/common/repository/TranscriptionDocumentRepository.java

```diff
@@ -1,9 +1,57 @@
 package uk.gov.hmcts.darts.common.repository;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 import uk.gov.hmcts.darts.common.entity.TranscriptionDocumentEntity;
+import uk.gov.hmcts.darts.transcriptions.model.TranscriptionDocumentResult;
+
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+import java.util.List;
 
 @Repository
 public interface TranscriptionDocumentRepository extends JpaRepository<TranscriptionDocumentEntity, Integer> {
-}
+
+    @Query("""
```

Comment:
@chrisbellingham-hmcts  a few things. The sub string searches on owner, requested by and courthouse display name has been made case insensitive

We left join here in order for the search to proceed under most eventualities. This does means that the hearing and case information may return null depending on the database data. Is this ok or do we a hard join?

Reply:
Also if there is no courthouse connected directly to the transcription as well as the transcription having no hearings then the courthouse would be returned as null also

Reply:
A `transcription` will generally be related to one or more `hearing` via `hearing_transcription_ae`. This will always be the scenario for modernised.

However for some migrated legacy data however, there may be no `transcription` -> `hearing` link. In this scenario there would be a link between the `transcription` and one or many `court_case`, via `case_transcription_ae`.

So I think in either scenario the `courthouse` should be resolvable via `court_case`, either via the `transcription` -> `hearing` > `court_case` route or the `transcription` > `court_case` route.

@hemantasharma1129 is that the correct assessment based on our recent discussions?

Reply:
Agree with @chrisbellingham-hmcts explanation. Dont we have the method "getCourtHouse" already in TranscriptionEntity to fetch the courthouse which will cater for bot association table? 
@mestebanez

Reply:
ok so by the looks of it transcription.courtroom is not needed to ascertain the court house. I will change.

Reply:
we do have convenience methods for getting a courthouse and getting a court case but they are not useable in hql also they only give one record in the case of multiple existing. I have changed the query as per @chrisbellingham-hmcts comments. Hopefully this aligns